cmd/eval_test.go

```diff
@@ -1543,3 +1544,150 @@ func TestBundleWithStrictFlag(t *testing.T) {
 	}
 
 }
+
+func TestEvalDiscardOutput(t *testing.T) {
+	query := "1*2+3"
+	tests := []struct {
+		format, expected string
+	}{
+		{
+			format: evalDiscardOutput,
+			expected: `{
+  "result": "discarded"
+}
+`},
+	}
+	params := newEvalCommandParams()
+	err := params.outputFormat.Set(evalDiscardOutput)
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+	var buf bytes.Buffer
+	_, err = eval([]string{query}, params, &buf)
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+	if actual := buf.String(); actual != tests[0].expected {
+		t.Errorf("expected output %q\ngot %q", tests[0].expected, actual)
+	}
+}
+
+func TestEvalDiscardNILOutput(t *testing.T) {
+	query := "1/0"
+	tests := []struct {
+		format, expected string
+	}{
+		{
+			format: evalDiscardOutput,
+			expected: `{}
+`},
+	}
+	params := newEvalCommandParams()
+	err := params.outputFormat.Set(evalDiscardOutput)
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+	var buf bytes.Buffer
+	_, err = eval([]string{query}, params, &buf)
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+	if actual := buf.String(); actual != tests[0].expected {
+		t.Errorf("expected output %q\ngot %q", tests[0].expected, actual)
+	}
+}
+
+func TestEvalDiscardShowBuiltinErrors(t *testing.T) {
```

Comment:
suggestion: since we have lists of tests (but only one item in them), perhaps we  can merge this test with the previous example like this:

```go
func TestEvalDiscardOutput(t *testing.T) {
	tests := map[string]struct {
		query, format, expected string
		params                  evalCommandParams
	}{
		"success example": {
			query: "1*2+3",
			params: func() evalCommandParams {
				params := newEvalCommandParams()
				err := params.outputFormat.Set(evalDiscardOutput)
				if err != nil {
					t.Fatalf("unexpected error: %s", err)
				}
				return params
			}(),
			expected: `{
  "result": "discarded"
}
`},
		"error example": {
			query: "1/0",
			params: func() evalCommandParams {
				params := newEvalCommandParams()
				err := params.outputFormat.Set(evalDiscardOutput)
				if err != nil {
					t.Fatalf("unexpected error: %s", err)
				}
				return params
			}(),
			expected: `{
  "result": "discarded"
}
`},
	}

	for name, tc := range tests {
		t.Run(name, func(t *testing.T) {
			var buf bytes.Buffer
			_, err := eval([]string{tc.query}, tc.params, &buf)
			if err != nil {
				t.Fatalf("unexpected error: %s", err)
			}
			if actual := buf.String(); actual != tc.expected {
				t.Errorf("expected output %q\ngot %q", tc.expected, actual)
			}
		})
	}
}

```

(haven't tested this, but hopefully it explains how the table [tests might](https://dev.to/boncheff/table-driven-unit-tests-in-go-407b) work)

Reply:
For the second case having query `"1/0"`, it should have `params.showBuiltinErrors = true` right?

Reply:
Ahh, yes! I missed that from my example! 😎 good spot...